src/tests/unit/platform/test_policy_ruleconfigs.py

```diff
@@ -0,0 +1,296 @@
+    # Replace all rule configs with the base class for purposes of this test
+    cfgs = policy._info.get("rule_configs", [])
+    ruleconfigobjects = [PolicyRuleConfig(api, policy, cfg['id'], cfg, force_init=False, full_doc=True) for cfg in cfgs]
+    policy._object_rule_configs = dict([(rconf.id, rconf) for rconf in ruleconfigobjects])
+    policy._object_rule_configs_need_load = False
+    # proceed with test
+    rule_config = random.choice(list(policy.object_rule_configs.values()))
+    old_name = rule_config.name
+    old_category = rule_config.category
+    old_parameters = rule_config.parameters
+    rule_config.refresh()
+    assert rule_config.name == old_name
+    assert rule_config.category == old_category
+    assert rule_config.parameters == old_parameters
+
+
+def test_rule_config_add_base_not_implemented(cbcsdk_mock):
+    """Verifies that adding a new BaseRuleConfig is not implemented."""
+    cbcsdk_mock.mock_request('GET', '/policyservice/v1/orgs/test/policies/65536/configs/presentation',
+                             POLICY_CONFIG_PRESENTATION)
+    api = cbcsdk_mock.api
+    policy_data = copy.deepcopy(FULL_POLICY_1)
+    rule_config_data1 = [p for p in enumerate(policy_data['rule_configs'])
+                         if p[1]['id'] == '88b19232-7ebb-48ef-a198-2a75a282de5d']
+    rule_config_data = rule_config_data1[0][1]
+    del policy_data['rule_configs'][rule_config_data1[0][0]]
+    policy = Policy(api, 65536, policy_data, False, True)
+    new_rule_config = PolicyRuleConfig(api, policy, '88b19232-7ebb-48ef-a198-2a75a282de5d', rule_config_data,
+                                       force_init=False, full_doc=True)
+    new_rule_config.touch()
+    with pytest.raises(NotImplementedError):
+        new_rule_config.save()
+
+
+def test_rule_config_delete_base_not_implemented(cbcsdk_mock):
+    """Verifies that deleting a BaseRuleConfig is not implemented."""
+    cbcsdk_mock.mock_request('GET', '/policyservice/v1/orgs/test/policies/65536/configs/presentation',
+                             POLICY_CONFIG_PRESENTATION)
+    api = cbcsdk_mock.api
+    policy = Policy(api, 65536, copy.deepcopy(FULL_POLICY_1), False, True)
+    # Replace all rule configs with the base class for purposes of this test
+    cfgs = policy._info.get("rule_configs", [])
+    ruleconfigobjects = [PolicyRuleConfig(api, policy, cfg['id'], cfg, force_init=False, full_doc=True) for cfg in cfgs]
+    policy._object_rule_configs = dict([(rconf.id, rconf) for rconf in ruleconfigobjects])
+    policy._object_rule_configs_need_load = False
+    # proceed with test
+    with pytest.raises(NotImplementedError):
+        policy.delete_rule_config('88b19232-7ebb-48ef-a198-2a75a282de5d')
+
+
+def test_rule_config_modify_by_base_method_invalid_id(cb):
+    """Tests modifying a PolicyRuleConfig object using replace_rule_config, but with an invalid ID."""
+    policy = Policy(cb, 65536, copy.deepcopy(FULL_POLICY_1), False, True)
```

Comment:
Instead of creating the policy in most of these tests why not make it fixture that resets with each execution?

Reply:
It was just as easy to copy the line of code that creates the policy.  Besides, in some of the tests, we use the `cb` fixture (because no calls are expected to actually be made to the server) and in some, we use `cbcsdk_mock` (because we're making a call to the server that needs a mock response). To do as you say, I'd need two separate fixtures.

Reply:
Please put this in a fixture to focus the test functions on the desired component that is being tested. The Policy object being created that way is a non standard practice we don't want to draw more attention to either

Reply:
Here is the desired fixture to be added so we can remove the Policy creation from the test functions
```
@pytest.fixture(scope="function")
def policy(monkeypatch, cb):
    """Mocks CBC SDK for unit tests"""
    return Policy(cb, 65535, copy.deepcopy(FULL_POLICY_1), False, True)
```

Reply:
This change has been pushed.